### PR TITLE
VIH-9260 - add Dynatrace Alerts to CVP

### PR DESCRIPTION
**Pull request by v1ferrarij**
### JIRA link (if applicable) ###

VIH-9260

### Change description ###

add Dynatrace Alerts to CVP

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

File: cloudconfig.tpl [hunk 1]
```diff
@@ -836,11 +836,69 @@ write_files:
...
+        project="${project}" 
+
+        # Set Dynatrace Details.
+        dynatrace_token="${dynatrace_token}"
+        dynatrace_tenant="${dynatrace_tenant}"
+
+        # Java Key Store Details.
+        jksPath="/usr/local/WowzaStreamingEngine/conf/ssl.wowza.jks"
+        wowzaServerPath="/usr/local/WowzaStreamingEngine/conf/Server.xml"
+        jksPass=$(cat $wowzaServerPath | grep -i KeyStorePassword | sed 's/<KeyStorePassword>//' | sed 's/<\/KeyStorePassword>//')
```
Comment (chris-pateman) [inline cloudconfig.tpl:862]: This can get passed in like the other vars `${certPassword}`

File: cloudconfig.tpl [hunk 2]
```diff
@@ -836,11 +836,69 @@ write_files:
...
 
         sudo -u wowza bash -c "crontab $cronTaskPath"
+  - owner: wowza:wowza
+    permissions: 0775
+    path: /home/wowza/check-cert.sh
+    content: |
+        #!/bin/bash
+
+        # Project
+        project="${project}"
```
Comment (chris-pateman) [inline cloudconfig.tpl:853]: project will always be CVP so do we need to pass this in?

File: modules/wowza/main.tf [hunk 3]
```diff
@@ -298,6 +301,12 @@ data "azurerm_key_vault_secret" "ssh_pub_key" {
   key_vault_id = data.azurerm_key_vault.cvp_kv.id
 }
 
+data "azurerm_key_vault_secret" "dynatrace_token" {
```
Comment (chris-pateman) [inline modules/wowza/main.tf:304]: I would add the token in via TF as well. Put it in the ADO Library and pull it into TF to add into the vault or just straight into the script

File: modules/wowza/main.tf [hunk 4]
```diff
@@ -275,6 +275,9 @@ data "template_file" "cloudconfig" {
     keyVaultName            = "cvp-${var.env}-kv"
     domain                  = "cvp-recording.${local.domain_dns_prefix}.platform.hmcts.net"
     wowzaVersion            = var.wowza_version
+    project                 = "CVP"
+    dynatrace_tenant        = var.env == "prod" ? "ebe20728" : "yrk32651"
```
Comment (chris-pateman) [inline modules/wowza/main.tf:279]: This is a var in the ADO pipeline already so can be passed into the TF

**Review by chris-pateman — CHANGES_REQUESTED** [review]

File: cloudconfig.tpl (hunk 1 above)
Comment (v1ferrarij) [reply cloudconfig.tpl:862]: I think this is a better way, then we never need to pass in the JKS password and will never need to update in our code, it is always being retrieved from the Server file which should always have the correct password.

File: cloudconfig.tpl (hunk 2 above)
Comment (v1ferrarij) [reply cloudconfig.tpl:853]: true, was just templated for VH and CVP

File: cloudconfig.tpl (hunk 1 above)
Comment (chris-pateman) [reply cloudconfig.tpl:862]: its already always passed in here >
https://github.com/hmcts/cvp-audio-ingress/blob/master/modules/wowza/main.tf#L265

and  used:
https://github.com/hmcts/cvp-audio-ingress/blob/master/cloudconfig.tpl#L27
https://github.com/hmcts/cvp-audio-ingress/blob/master/cloudconfig.tpl#L159
https://github.com/hmcts/cvp-audio-ingress/blob/master/cloudconfig.tpl#L859

File: pipeline/steps/az-shared-config.yml [hunk 5]
```diff
@@ -171,6 +171,32 @@ steps:
...
+        $kvName="$(kvName)"
+
+        Write-Host "##[warning]!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
+        Write-Host "##[warning]THIS SCRIPT WILL ONLY WORK WITH NON-SECRET VARIABLES!"
+        Write-Host "##[warning]!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
+        
+        echo $(System.AccessToken) | az devops login --organization $organisation
+        $groupId=$(az pipelines variable-group list --org $organisation --project "$project" --query "[?name=='$groupName'].{id:id}[0]" -o tsv --only-show-errors)
+
+        $secrets=$(az pipelines variable-group variable list --org $organisation --project $project --id $groupId) | ConvertFrom-Json -AsHashTable
```
Comment (chris-pateman) [inline pipeline/steps/az-shared-config.yml:194]: The Dynatrace token is a secret value, so should not be plain text in the ADO or passed here plainly

**Review by chris-pateman — CHANGES_REQUESTED** [review]

**Review by chris-pateman — DISMISSED** [review]
LGTM

**Review by chris-pateman — APPROVED** [review]
LGTM

**Pull request merged**